test/models/work_test.rb

```diff
@@ -0,0 +1,153 @@
+      new_book = Work.new(title: "new book", category: 'book')
+      new_book.must_be :valid?
+
+      # can create a new work with category 'album'
+      new_album = Work.new(title: "new album", category: 'album')
+      new_album.must_be :valid?
+
+      # can create a new work with category 'movie'
+      new_movie = Work.new(title: "new movie", category: 'movie')
+      new_movie.must_be :valid?
+
+      # CANNOT create a new work with any other category
+      new_invalid_work = Work.new(title: "new album", category: 'other')
+      new_invalid_work.wont_be :valid?
+    end # requires album, book, or movie as category
+  end # validations
+
+  describe "methods" do
+    describe "sort_by_vote_count" do
+      it "will sort the works by vote count in decending order" do
+        #TODO: figure out why this test isn't passing!
+        # Arrange
+        # create a second work to vote on
+        w3.save
+        w.save
+        # create a second user to vote
+        u2 = User.last
+
+        # Act (vote on the two works, w and w3)
+        w.votes.create(user_id: u.id, work_id: w.id)
+        w3.votes.create(user_id: u2.id, work_id: w3.id)
+        w3.votes.create(user_id: u.id, work_id: w3.id)
+        sorted = Work.sort_by_vote_count
+
+        # Assert
+        sorted.first.id.must_equal w3.id
+      end # it will sort by vote_count
+    end # sort_by_vote_count
+
+    describe "top_work" do
+      it "will return the top rated work" do
+        w3.save
+        w4.save
+        u2 = User.last
+
+        w3.votes.create(user_id: u.id, work_id: w3.id)
+        w3.votes.create(user_id: u2.id, work_id: w3.id)
+        w4.votes.create(user_id: u.id, work_id: w4.id)
+        top = Work.top_work
+
+        top.title.must_equal w3.title
+      end # return the top work
+
+      it "will return the the first work if there is a tie for the most votes" do
+        w3.save
+        w4.save
+        u2 = User.last
+
+        w3.votes.create(user_id: u.id, work_id: w3.id)
+        w3.votes.create(user_id: u2.id, work_id: w3.id)
+        w4.votes.create(user_id: u.id, work_id: w4.id)
+        w4.votes.create(user_id: u2.id, work_id: w4.id)
+        top = Work.top_work
+
+        top.title.must_equal w3.title
+      end # return the first in a tie
+
+      it "will return the first work if no works have votes" do
+        # TODO: why does this test fail even though the titles match???
+        w4.save
+        w3.save
+
+        top = Work.top_work
+
+        top.title.must_equal w4.title
+      end # return when no votes
+
+    end # top_work
+
+
+    describe "top_ten" do
+      it "will return the top ten works of a category" do
+        ten_works = Work.top_ten("album")
+
+        length = ten_works.length
+        i = 0
+        while i < length
+          ten_works.votes[i].count.must_be :>=, ten_work.votes[i - 1].count
+          ten_works[i].category.must_equal 'album'
+        end
+      end # return the top ten works for a category
```

Comment:
What if there are no works? What if there are more than 10 works? What if there are no votes, or if there's a tie between the 10th and 11th work?